### PR TITLE
TSDK-789 BTC Monitoring

### PR DESCRIPTION
**Pull request by mundacho**
## Purpose

The goal of this PR is to implement monitoring of the BTC network into the bridge.

## Approach

- A new architecture was implemented using a state machine and streams to keep the state of the system in sync with the sessions.
- `/api/confirm-deposit-btc` WS was removed.
- Added new parameters for the launcher:  `--btc-url`, `--btc-port`, `--btc-user`, `--btc-password`, `--zmq-url`, `--zmq-port`. 
- Added support for configuring these parameters using environment variables (for docker deploys)
- Modified test to rely on monitoring.

## Testing

- The unit tests were updated
- The integration test was updated
- Further testing is required for UI, will be done in another ticket.

## Tickets

* TSDK-789

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## BTC Regtest docker file
+
+This docker file is used to create a BTC regtest node. It is very simple
+and just hardcodes the parameters exactly as what we need for our tests.
+It is published in docker hub as `jimetoplprotocol/bitcoin-zmq:v25-regtes`.
```
Comment (DiademShoukralla) [inline docker/README.md:5]:
```suggestion
It is published in docker hub as `toplprotocol/bitcoin-zmq:v25-regtest`.
```

It looked like a typo to me. https://hub.docker.com/r/toplprotocol/bitcoin-zmq/tags

File: integration/src/test/scala/co/topl/bridge/BridgeIntegrationSpec.scala [hunk 2]
```diff
@@ -238,27 +253,28 @@ class BridgeIntegrationSpec extends CatsEffectSuite {
...
+        //       ).withContentType(
+        //         `Content-Type`.apply(MediaType.application.json)
+        //       ).withEntity(
+        //         SyncWalletRequest(
+        //           "secret"
+        //         )
+        //       )
+        //     )
+        //   })
+        // _ <- IO.println("syncWalletResponse: " + syncWalletResponse)
```
Comment (DiademShoukralla) [inline integration/src/test/scala/co/topl/bridge/BridgeIntegrationSpec.scala:276]: Were these comments meant to be left in?

**Review by DiademShoukralla — APPROVED** [review]
LGTM

File: integration/src/test/scala/co/topl/bridge/BridgeIntegrationSpec.scala (hunk 2 above)
Comment (mundacho) [reply integration/src/test/scala/co/topl/bridge/BridgeIntegrationSpec.scala]: I removed them, thanks!

**Pull request merged**